### DISCUSSION TITLE
Duplicate multiple response headers

### DISCUSSION THREAD

**Discussion by cp-weng-tkspring** [fastapi/fastapi#10299] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi.responses import RedirectResponse

@router.get("/path")
async def operation():
    response = RedirectResponse(...)
    response.set_cookie("foo", ...)
    response.set_cookie("bar", ...)
    return response

# Expected Response Header
Set-Cookie: foo=...
Set-Cookie: bar=...

# Actual Response Header
Set-Cookie: foo=..., bar=...
```

### Description

FastAPI doesn't support duplicate response headers, especially for "Set-Cookie".
However, Chrome doesn't respect fastapi's comma-separated header style.
That makes my service totally broken when using Chrome.

### Operating System

macOS

### FastAPI Version

0.103.1

### Pydantic Version

2.3.0

### Python Version

Python 3.10.2

### Additional Context

There is one relative issue: https://github.com/encode/starlette/issues/650

**Comment by YuriiMotov** [comment, marked as answer]
It was probably fixed somewhere outside of FastAPI, but it works well with fastapi=0.103.1:

```python
from typing import Annotated
from fastapi import Cookie, FastAPI
from fastapi.responses import RedirectResponse


app = FastAPI()


@app.get("/path")
async def operation():
    response = RedirectResponse("/path2")
    response.set_cookie("foo", "foo-value")
    response.set_cookie("bar", "bar-value")
    return response


@app.get("/path2")
async def operation2(foo: Annotated[str, Cookie()], bar: Annotated[str, Cookie()]):
    return {"foo": foo, "bar": bar}
```


```
HTTP/1.1 307 Temporary Redirect
date: ***
server: uvicorn
content-length: 0
location: /path2
set-cookie: foo=foo-value; Path=/; SameSite=lax
set-cookie: bar=bar-value; Path=/; SameSite=lax
```
